src/Database/QueryCompiler.php

```diff
@@ -351,7 +351,9 @@ protected function _buildSetPart(array $parts, Query $query, ValueBinder $binder
     protected function _buildUnionPart(array $parts, Query $query, ValueBinder $binder): string
     {
         $parts = array_map(function ($p) use ($binder) {
-            $p['query'] = $p['query']->sql($binder);
+            if (!is_string($p['query'])) {
+                $p['query'] = $p['query']->sql($binder);
+            }
```

Comment:
Can this still happen, the query already being an SQL string?

Reply:
1st argument `union()` of can be a string, so it's possible, we just don't have a test case for this case currently.

Reply:
D'oh! Right in front of my face :)